packages/editor/src/plugins/image/components/pixabay-image-search.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { ChangeEvent, useState, useEffect } from 'react'
+
+import { useEditorStrings } from '@/contexts/logged-in-data-context'
+import { cn } from '@/helper/cn'
+
+interface PixabayResponse {
+  total: number
+  totalHits: number
+  hits: PixabayImage[]
+}
+interface PixabayImage {
+  id: number
+  webformatURL: string
+  tags: string
+  previewURL: string
+  largeImageURL: string
+}
+
+interface PixabayImageSearchProps {
+  onSelectImage: (imageUrl: string) => void
+}
+
+export const PixabayImageSearch = ({
+  onSelectImage,
+}: PixabayImageSearchProps) => {
+  const [query, setQuery] = useState('')
+  const [images, setImages] = useState<PixabayImage[]>([])
+  const [isLoadingImage, setIsLoadingImage] = useState(false)
+  const [isSearching, setIsSearching] = useState(false)
+
+  const [showTags, setShowTags] = useState(true)
+
+  const editorStrings = useEditorStrings()
+  const imageStrings = editorStrings.plugins.image
+  const apiKey = 'secret'
+
+  useEffect(() => {
+    if (isLoadingImage) {
+      document.body.classList.add('overflow-hidden')
+    } else {
+      document.body.classList.remove('overflow-hidden')
+    }
+  }, [isLoadingImage])
+
+  const handleSearch = async (searchQuery: string) => {
+    if (!searchQuery) return
+    setIsSearching(true)
+    try {
+      const response = await fetch(
+        `https://pixabay.com/api/?key=${apiKey}&q=${encodeURIComponent(searchQuery)}&image_type=photo`
+      )
+      const data = (await response.json()) as PixabayResponse
+      setImages(data.hits)
+      setShowTags(false)
```

Comment:
currently I think showTags can be calculated by the value of `images` and does not need to be an own state, is that correct?

Reply:
So far I've looked at it more as the "front search page" with the tags, and the "search results" page with the images, and if 0 images are returned then we are informed with text, but the tags are not present

Reply:
My point is that you should not put stuff in the state that can be calculated by existing state.
Maybe I still don't understand correctly but I still think that might be happening here.

Reply:
I've updated the code, please let me know what you think